cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -124,6 +125,15 @@ class LeadImpl extends ServerImpl with Lead with Logging {
         conf.set(key, v)
       })
 
+      /**
+       * If kryoserializer is being used then register snappyclasses for kryoserializer.
+       * The snappyclasses will not get registered if user is creating its own spark context.
+       * In that case user has to specify the snappy kryo registrator externally
+       */
+      if (conf.contains("spark.serializer") && conf.get("spark.serializer").equals("org.apache.spark.serializer.KryoSerializer")) {
+        conf.set("spark.kryo.registrator", getSnappyKryoRegistrarClass())
```

Comment:
@rishitesh Please correct me if i am wrong, as far as i know user can only specify registrator if he is creating the `SparkConf/SparkContext` by himself which we are not recommending right?. As mentioned in my comments in `LeadImpl.scala` if user is creating his own SparkContext then he has to register our registartor which he can extend to register his own classes or he can use `conf.registerKryoClasses`  method